### PR TITLE
Add "argon" note and stage design for osu!mania

### PR DESCRIPTION
**Pull request by peppy**
Going to PR this as-is as the bulk of the code can be reviewed and merged without issue. Fine tuning is an ongoing effort.

Pieces which still need fine tuning:

- Colouring is a bit too saturated after removing key area local background. Will tweak this as we go.
- Hold note "ticks" are not skinned. Weirdly, this doesn't even support legacy skinning as it stands and uses a single visual for all skins.
- Hold note tracks are a bit hard to see. Will iterate on this.

Also:

- [x] Depends on https://github.com/ppy/osu/pull/20592

https://user-images.githubusercontent.com/191335/194477996-8acefe9c-95b2-4c7f-a6e6-181a6c69ee2d.mp4

### CODE REVIEW

**Comment by smoogipoo** [comment]
Without having merged the prereq yet, something may have regressed in this branch because i'm getting all black colours?

**Comment by smoogipoo** [comment]
Fixed after merge :shrug:

**Comment by peppy** [comment]
> Without having merged the prereq yet, something may have regressed in this branch because i'm getting all black colours?

See https://github.com/ppy/osu/pull/20592#issuecomment-1271178529, would happen if you had beatmap skins on, and was fixed as a commit earlier today 😅 .

File: osu.Game/Skinning/LegacyManiaSkinConfigurationLookup.cs [hunk 1]
```diff
@@ -42,6 +42,8 @@ public enum LegacyManiaSkinConfigurationLookups
         HitPosition,
         ScorePosition,
         LightPosition,
+        StagePaddingTop,
+        StagePaddingBottom,
```
Comment (smoogipoo) [inline osu.Game/Skinning/LegacyManiaSkinConfigurationLookup.cs:46]: Yikes...

**Review by smoogipoo — APPROVED** [review]

File: osu.Game/Skinning/LegacyManiaSkinConfigurationLookup.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Skinning/LegacyManiaSkinConfigurationLookup.cs:46]: This was the path of least resistance to make this work.

I'm not sure how these will be handled in the future, but I think we'll want to move *all* of this configuration to `SourceSettings` on respective components so they can be edited in the in-game editor. That's going to have to be done as a batch operation anyway, so I don't see adding new entries here as such a bad thing until then.

**Pull request merged**